medium/users/services.py

```diff
@@ -0,0 +1,38 @@
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth.password_validation import validate_password
+from django.core.validators import validate_email
+
+from medium.users.models import User
+
+
+class UserService:
+    def create_user(self, name: str, email: str, password: str) -> User:
+        user: User = User(name=name)
+        validate_email(email)
+        user.email = email
+
+        validate_password(password, user)
+        hashed_password: str = make_password(password)
```

Comment:
after validating the fields
I recommend using `create_user` from `BaseUserManager`

Reply:
Please, send me a reference for that, I searched the codebase for this method and I didn't find it

Reply:
https://docs.djangoproject.com/en/4.0/ref/contrib/auth/#manager-methods

Reply:
I can't call the `create_user` method directly because the `validate_password` method needs to take the user object to run some validators against it and if I called the `create_user` method to get the object it will create the object and save it directly without validating it.

What the `create_user` method does is: 
1. Normalizing the `username` and the `email`.
2. Creating a user object. 
3. Hashing the password.
4. Saving the object and returning it.

To solve this, instead of directly calling the `create_user` method I will call the `username` and `email` normalizer methods from the user service directly